src/front/wgsl.rs

```diff
@@ -466,6 +468,34 @@ impl Parser {
         }
     }
 
+    fn deconstruct_composite_type(
+        type_arena: &mut Arena<crate::Type>,
+        ty: Handle<crate::Type>,
+        index: usize,
+    ) -> Result<Handle<crate::Type>, Error<'static>> {
+        let ty = match type_arena[ty].inner {
+            crate::TypeInner::Vector {
+                ref kind,
```

Comment:
you don't need to take `kind` and `width` by reference, they are small and copyable

Reply:
Yeah, I attempted writing that match again, and it now works out normally with `ref`s and all that.